jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushTagToRepositoryAction.kt

```diff
@@ -0,0 +1,174 @@
+import com.intellij.docker.runtimes.DockerApplicationRuntime
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.remoteServer.impl.configuration.RemoteServerImpl
+import com.intellij.remoteServer.runtime.ServerConnectionManager
+import com.intellij.remoteServer.runtime.ServerConnector
+import com.intellij.remoteServer.runtime.deployment.ServerRuntimeInstance
+import com.intellij.ui.CollectionComboBoxModel
+import com.intellij.ui.SimpleListCellRenderer
+import com.intellij.ui.layout.listCellRenderer
+import com.intellij.ui.layout.panel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.jetbrains.concurrency.AsyncPromise
+import org.jetbrains.concurrency.await
+import software.amazon.awssdk.services.ecr.EcrClient
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.services.ecr.EcrPushRequest
+import software.aws.toolkits.jetbrains.services.ecr.getDockerLogin
+import software.aws.toolkits.jetbrains.services.ecr.resources.EcrResources
+import software.aws.toolkits.jetbrains.services.ecr.toLocalImageList
+import software.aws.toolkits.jetbrains.ui.ResourceSelector
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.ui.blankAsNull
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.resources.message
+
+class PushTagToRepositoryAction :
+    DumbAwareAction(message("ecr.push.action")),
+    CoroutineScope by ApplicationThreadPoolScope("PushRepositoryAction") {
+    private val dockerServerRuntime: Deferred<DockerServerRuntimeInstance> =
+        async(start = CoroutineStart.LAZY) { getDockerServerRuntimeInstance() }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.getRequiredData(LangDataKeys.PROJECT)
+        val client: EcrClient = project.awsClient()
+
+        val dialog = PushToEcrDialog(project, dockerServerRuntime)
+        val result = dialog.showAndGet()
+
+        if (!result) {
+            // user cancelled; noop
+            return
+        }
+        val pushRequest = dialog.getPushRequest()
+
+        launch {
+            val authData = withContext(Dispatchers.IO) {
+                client.authorizationToken.authorizationData().first()
+            }
+
+            val (username, password) = authData.getDockerLogin()
+            val model = DockerRepositoryModel().also {
+                val repoUri = pushRequest.remoteRepo.repositoryUri
+                // fix
```

Comment:
What is there to fix here?

Reply:
missed comment, basically un-hardcode all the values here